compiler/rustc_ast_lowering/src/item.rs

```diff
@@ -86,7 +86,7 @@ impl<'a, 'hir> ItemLowerer<'a, 'hir> {
             impl_trait_defs: Vec::new(),
             impl_trait_bounds: Vec::new(),
             allow_try_trait: Some([sym::try_trait_v2, sym::yeet_desugar_details][..].into()),
-            allow_gen_future: Some([sym::gen_future][..].into()),
+            allow_gen_future: Some([sym::gen_future, sym::closure_track_caller][..].into()),
```

Comment:
I wonder if it makes sense to split `gen_future` and `closure_track_caller` into separate entires?

I think it's fine like this, since `allow_gen_future` is only used in one place and since we want track caller to work we also need the `closure_track_caller` feature. Separating them seems like extra work with little benefit.